src/oc_erchef/apps/chef_objects/src/chef_cookbook_version.erl

```diff
@@ -589,6 +672,84 @@ list_query(_ObjectRec) ->
 bulk_get_query(_ObjectRec) ->
     error(not_implemented).
 
+%% for v2 metadata, the "segment" is encoded in to the name of file
+%% so for data that is v0 on disk we have to prepend the segment to the name
+add_segment_to_filename(Segment, File) ->
+    Fn = ej:get({<<"name">>}, File),
+    Fn1 = iolist_to_binary([Segment, "/", Fn]),
+    ej:set({<<"name">>}, File, Fn1).
+
+remove_segment_from_filename(File) ->
+    [Segment | Name ] = get_segment_from_record(File),
+    case Name of
+        % if Name is nil, then we have a root file (like metadata.rb) - so we'll set the name to the segment, and the segment to <<"root_files">>
+        [] ->
+            { <<"root_files">>, ej:set({<<"name">>}, File, Segment)};
+        _ ->
+            Record = ej:set({<<"name">>}, File, lists:last(Name)),
+            { Segment, Record }
+    end.
+
+populate_all_files(Segment, Data, Metadata) ->
+    lists:foldl(fun(File, CB) ->
```

Comment:
Since Segment is a constant, It might be easier to invert the order; e.g. 
```
    ExpandedFiles = case Segment of 
        <<"root_files">> -> Data;
    _ ->
         [ add_segment_to_filename(Segment, File) || File <- Data ]
    end,
    ej:set({<<"all_files">>}, Metadata, ExpandedFiles)
```